test/Migrator/wrap_optional.swift

```diff
@@ -36,19 +36,15 @@ class MyExtraCities : ExtraCities {
 }
 
 typealias IntAnd<T> = (Int, T)
-class Outer {
-  typealias Inner = (String?, String) -> String!
-}
 
 class MyExtraCitiesWithAliases : ExtraCities {
-  func blibli(x: Outer.Inner) {}
+  func blibli(x: (String?, String) -> String!) {}
```

Comment:
Why is this allowed but the typealias form is not? I would expect a function type would never be able to have an IUO return, only a function *declaration* would

Reply:
Yeah this was an oversight. This was getting caught at one point but I tweaked some code that resulted in this being accepted again and I didn't have a separate test confirming the diagnostic. I'll add one.